RFS/src/testFixtures/java/com/rfs/framework/PreloadedDataContainerOrchestrator.java

```diff
@@ -0,0 +1,285 @@
+        var httpClient = new ApacheDockerHttpClient.Builder()
+                .dockerHost(config.getDockerHost())
+                .sslConfig(config.getSSLConfig())
+                .maxConnections(100)
+                .connectionTimeout(Duration.ofSeconds(30))
+                .responseTimeout(Duration.ofSeconds(45))
+                .build();
+
+        return DockerClientImpl.getInstance(config, httpClient);
+    }
+
+    private static Image getExistingImage(DockerClient dockerClient, String imageName, String tag,
+                                          boolean pullIfUnavailable) throws InterruptedException {
+        var fullName = formatFullImageName(imageName, tag);
+        for (var image : dockerClient.listImagesCmd().exec()) {
+            for (String s : image.getRepoTags()) {
+                if (fullName.equals(s)) {
+                    return image;
+                }
+            }
+        }
+        if (pullIfUnavailable) {
+            log.warn("Image not found. Pulling image: " + fullName);
+            dockerClient.pullImageCmd(imageName)
+                    .withTag(tag)
+                    .exec(new PullImageResultCallback())
+                    .awaitCompletion(PULL_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            return getExistingImage(dockerClient, imageName, tag, false);
+        }
+        return null;
+    }
+
+    @NotNull
+    private static String formatFullImageName(String imageName, String tag) {
+        return imageName + ":" + tag;
+    }
+
+    private void makeNewImage(DockerClient dockerClient, String imageName, String tag)
+            throws IOException, InterruptedException
+    {
+        var imageResponse = dockerClient.inspectImageCmd(baseSourceVersion.imageName).exec();
+        var originalEntrypoint = imageResponse.getConfig().getEntrypoint();
+        var originalCmd = imageResponse.getConfig().getCmd();
+        final var replacementCommand = Streams.concat(
+                        Stream.of("/bin/sh", "-c"),
+                        Stream.of(Streams.concat(
+                                Stream.of("printenv && "),
+                                        Optional.ofNullable(originalEntrypoint).stream().flatMap(Arrays::stream),
+                                        Optional.ofNullable(originalCmd).stream().flatMap(Arrays::stream),
+                                        Stream.of("&", "tail", "-f", "/dev/null"))
+                                .collect(Collectors.joining(" "))))
+                .toArray(String[]::new);
+
+        try (var network = Network.newNetwork();
+             var serverContainer = new ElasticsearchContainer(baseSourceVersion)
+                     .withNetwork(network)
+                     .withNetworkAliases(serverNameAlias)
+                     .withCreateContainerCmdModifier(cmd -> cmd.withEntrypoint(replacementCommand))
+                     .withStartupTimeout(Duration.ofDays(1))) {
```

Comment:
That is too long, maybe 1-5 minute?

Reply:
whoops!  Thanks.  1 day was REALLY useful so that I could keep the container alive to run experiments for an hour, etc.